build_web_compilers/build.yaml

```diff
@@ -132,16 +132,19 @@ builders:
         exclude:
           - test/**.node_test.dart
           - test/**.vm_test.dart
-      options:
-        dart2js_args:
-          - --minify
       dev_options:
-        dart2wasm_args:
-          - --enable-asserts
-        dart2js_args:
-          - --enable-asserts
+        compilers:
+          dartdevc:
+            extension: '.dart.js'
       release_options:
-        compiler: dart2wasm+dart2js
+        compilers:
```

Comment:
@sigmundch @kevmoo not sure who to talk to about this, but what do we want the defaults to be here? This controls the default release build configuration for all dart web apps.

Reply:
Just to clarify, do you mean the default output extensions for each compilation? default flags? both?

Reply:
> Just to clarify, do you mean the default output extensions for each compilation? default flags? both?

Both - we'd compile with `dart2js` (which we instruct to emit a `.dart2js.js` file here, we don't rename additional files for deferred imports), `dart2wasm` (outputs `.wasm`, `.dart2wasm.mjs`) and a custom entrypoint (`.dart.js`) running feature detection for WasmGC and then loading either the `.dart2js.js` script or the `.dart2wasm.mjs` module.

> This controls the default release build configuration for all dart web apps.

Although I suspect most users do override the default here - `build_web_compilers` never passed any optimization flags to `dart2js` by default.

Reply:
A downside of dart2wasm by default is that existing apps importing `dart:html` will see failed builds, so maybe we should use `dart2js` for now and enable both compilers in a future release after the legacy web interop libraries have been removed from the SDK?

Reply:
Yeah, I share that concern - unless we can determine upfront that the app has no uses of `dart:html`, then we could have an adaptive default.

That said, do we also offer a mode that is dart2wasm only? or only the combo wasm+js? Given that browser support is not there yet for wasm-gc, I'd prefer that we only offer the combo, so JS can be used as fallback for those browsers.

Reply:
> That said, do we also offer a mode that is dart2wasm only? 

We do as of https://github.com/dart-lang/build/pull/3727. This PR adds support for enabling multiple compilers.

> I'd prefer that we only offer the combo, so JS can be used as fallback for those browsers.

Personally I would like to be able to compile with dart2wasm only for testing purposes (e.g. in integration tests that should run with dart2js once and then with dart2wasm). These tests failing when running on unsupported browsers is better than testing dart2js twice. Perhaps we can emit a warning in the builder when only dart2wasm is enabled?

Reply:
Ok, lets use dart2js by default, and keep the other options as opt in